src/test/java/com/app/onlinebookstore/repository/book/BookRepositoryTest.java

```diff
@@ -0,0 +1,228 @@
+package com.app.onlinebookstore.repository.book;
+
+import com.app.onlinebookstore.model.Book;
+import com.app.onlinebookstore.model.Category;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class BookRepositoryTest {
+    private static Map<Long, Category> categories;
+    private static Map<Long, Book> books;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @BeforeAll
+    static void beforeAll() {
+        Category category1 = new Category(1L, "Classic novel", "A classic novel", false);
+        Category category2 = new Category(2L, "Dystopian novel", "A dystopian novel", false);
+        Category category3 = new Category(3L, "Romantic novel", "A romantic novel", false);
+        Category additional = new Category(4L, "Additional", "Additional category", false);
+
+        categories = Map.of(1L, category1, 2L, category2, 3L, category3, 4L, additional);
+
+        Book book1 = new Book(1L,
+                "To Kill a Mockingbird",
+                "Harper Lee",
+                "9780061120084",
+                BigDecimal.valueOf(9.99),
+                "A classic novel",
+                "/book1.jpg",
+                false,
+                Set.of(category1));
+
+        Book book2 = new Book(2L,
+                "1984",
+                "George Orwell",
+                "9780451524935",
+                BigDecimal.valueOf(10.54),
+                "A dystopian novel",
+                "/book2.jpg",
+                false,
+                Set.of(category2, additional));
+
+        Book book3 = new Book(3L,
+                "Pride and Prejudice",
+                "Jane Austen",
+                "9780486284736",
+                BigDecimal.valueOf(7.99),
+                "A romantic novel",
+                "/images/book3.jpg",
+                false,
+                Set.of(category3, additional));
+
+        books = Map.of(1L, book1, 2L, book2, 3L, book3);
+    }
+
+    @Test
+    @DisplayName("Find All Books by Category ID: When Category Exists, Return List of Books")
+    @Sql(scripts = {"classpath:sql-scripts/books/InsertImmutableBooks.sql"},
```

Comment:
you can consider extracting scripts location to the constants (even to the class with constants, but as you prefer), it would be really helpful when you decide to rename/move the script, so you'd need to change it only in one place